src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/IPathConverter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
+
+using System.Collections.Generic;
+using ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using System.Collections.ObjectModel;
+
+internal interface IPathConverter
+{
+    string? UpdatePath(string? path, PathConversionDirection updateDirection);
+
+    IEnumerable<string?> UpdatePaths(IEnumerable<string?> paths, PathConversionDirection updateDirection);
+
+    TestCase UpdateTestCase(TestCase testCase, PathConversionDirection updateDirection);
+
+    IEnumerable<TestCase> UpdateTestCases(IEnumerable<TestCase> testCases, PathConversionDirection updateDirection);
+
+    TestRunCompleteEventArgs UpdateTestRunCompleteEventArgs(TestRunCompleteEventArgs testRunCompleteEventArgs, PathConversionDirection updateDirection);
+
+    TestRunChangedEventArgs UpdateTestRunChangedEventArgs(TestRunChangedEventArgs testRunChangedArgs, PathConversionDirection updateDirection);
+
+    Collection<AttachmentSet> UpdateAttachmentSets(Collection<AttachmentSet> attachmentSets, PathConversionDirection updateDirection);
+
+    ICollection<AttachmentSet> UpdateAttachmentSets(ICollection<AttachmentSet> attachmentSets, PathConversionDirection updateDirection);
```

Comment:
Do we need both `Collection<T>` and `ICollection<T>`?

Reply:
Yes, they don't implicitly cast, I tried to remove it but it was requiring explicit cast which I don't want to do.

Reply:
That's funny doc says `ObjectModel.Collection` does implement `ICollection`. Anyways, marking as resolved. Thanks for trying it out.